pkg/resources/dynamic_table.go

```diff
@@ -200,6 +223,15 @@ func ReadDynamicTable(d *schema.ResourceData, meta interface{}) error {
 			return err
 		}
 	}
+	if strings.Contains(dynamicTable.Text, "initialize = 'ON_CREATE'") {
```

Comment:
Please set the refresh mode too. If I understand the docs correctly (https://docs.snowflake.com/en/sql-reference/sql/create-dynamic-table#required-parameters), it should be also extracted from the `Text` and not the `RefreshMode` prop.

Reply:
It was already set (line: 247)